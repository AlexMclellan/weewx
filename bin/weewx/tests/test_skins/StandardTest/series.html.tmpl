#errorCatcher Echo
## Copyright 2009-2021 Tom Keffer
## Distributed under terms of GPLv3.  See LICENSE.txt for your rights.
##
## This template tests tags used for generating series
##
##
## Specifying an encoding of UTF-8 is usually safe:
#encoding UTF-8
##
<!DOCTYPE HTML>
<html lang="en">
  <head>
    ## This choice should match the #encoding directive above
    <meta charset="UTF-8">
    <title>Current Weather Conditions</title>
    <link rel="stylesheet" type="text/css" href="weewx.css"/>
    <style>
        p { font-weight: bold; }
    </style>
  </head>

  <body>

    #set ($start, $stop, $data) = $month.outTemp.series(aggregation_type='max', aggregation_interval=86400)
    <p> For what follows, \$start, \$stop, and \$data are defined as:</p>
	<pre>\#set (\$start, \$stop, \$data) = \$month.outTemp.series(aggregation_type='max', aggregation_interval=86400)</pre>

	<hr/>
	<p>\$data:</p>
	<pre> $data </pre>

	<hr/>
	<p>\$data.format("%.2f"): </p>
	<pre>$data.format("%.2f")</pre>

	<hr/>
	<p>\$data.degree_C:</p>
	<pre>$data.degree_C</pre>

	<hr/>
	<p>\$data.json: </p>
	<pre>$data.json</pre>

	<hr/>
	<p>\$data.degree_C.json: </p>
	<pre>$data.degree_C.json</pre>

	<hr/>
	<p>\$data.raw:</p>
    <pre>$data.raw</pre>

	<hr/>
	<p>Aggregated series: \$month.outTemp.series(aggregation_type='max', aggregation_interval=86400).json():</p>
    <pre>$month.outTemp.series(aggregation_type='max', aggregation_interval=86400).json()</pre>

	<hr/>
	<p>Using shortcut 'day': \$month.outTemp.series(aggregation_type='max', aggregation_interval='day').json():</p>
    <pre>$month.outTemp.series(aggregation_type='max', aggregation_interval='day').json()</pre>

	<hr/>
	<p>Order by column: \$month.outTemp.series(aggregation_type='max', aggregation_interval=86400).json(order_by="column"):</p>
    <pre>$month.outTemp.series(aggregation_type='max', aggregation_interval=86400).json(order_by="column")</pre>

	<hr/>
	<p>Unaggregated series: \$day.outTemp.series.json():</p>
    <pre>$day.outTemp.series.json()</pre>

	<hr/>
	<p>Unaggregated series, just emit start times: \$day.outTemp.series.json(time_series='start'):</p>
    <pre>$day.outTemp.series.json(time_series='start')</pre>

    <hr/>
    <p>Unaggregated series, as a formatted string (<em>not JSON</em>) : \$day.outTemp.series:</p>
    <pre>$day.outTemp.series</pre>

    <hr/>
    <p>Unaggregated series, by column, as a formatted string (<em>not JSON</em>) \$day.outTemp.series.format(order_by='column'):</p>
    <pre>$day.outTemp.series.format(order_by='column')</pre>

    <hr/>
    <h2>Complex conversions</h2>

    <p>Starting series: \$month.windvec.series(aggregation_type='max', aggregation_interval='day').json()</p>
    <pre>
    $month.windvec.series(aggregation_type='max', aggregation_interval='day').json()
    </pre>

    <p> X-component: \$month.windvec.series(aggregation_type='max', aggregation_interval='day').x.json()</p>
    <pre>
    $month.windvec.series(aggregation_type='max', aggregation_interval='day').x.json()
    </pre>

    <p> x-component, in knots: \$month.windvec.series(aggregation_type='max', aggregation_interval='day').x.knot.json()</p>
    <pre>
    $month.windvec.series(aggregation_type='max', aggregation_interval='day').x.knot.json()
    </pre>

    <p>Y-component: \$month.windvec.series(aggregation_type='max', aggregation_interval='day').y.json()</p>
    <pre>
    $month.windvec.series(aggregation_type='max', aggregation_interval='day').y.json()
    </pre>

    <p>magnitude: \$month.windvec.series(aggregation_type='max', aggregation_interval='day').magnitude.json()</p>
    <pre>
    $month.windvec.series(aggregation_type='max', aggregation_interval='day').magnitude.json()
    </pre>

    <p>direction: \$month.windvec.series(aggregation_type='max', aggregation_interval='day').direction.json()</p>
    <pre>
    $month.windvec.series(aggregation_type='max', aggregation_interval='day').direction.json()
    </pre>

    <p>polar: \$month.windvec.series(aggregation_type='max', aggregation_interval='day').polar.json()</p>
    <pre>
    $month.windvec.series(aggregation_type='max', aggregation_interval='day').polar.json()
    </pre>

    <p>polar in knots: \$month.windvec.series(aggregation_type='max', aggregation_interval='day').polar.knot.json()</p>
    <pre>
    $month.windvec.series(aggregation_type='max', aggregation_interval='day').polar.knot.json()
    </pre>


  </body>

</html>
