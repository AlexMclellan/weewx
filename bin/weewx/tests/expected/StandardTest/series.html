<!DOCTYPE HTML>
<html lang="en">
  <head>
      <meta charset="UTF-8">
      <title>Current Weather Conditions</title>
      <style>
          p {
              font-weight: bold;
          }
          table, th, td {
              border: 1px solid black;
              border-collapse: collapse;
          }    </style>
  </head>

  <body>
    <h2>Unaggregated series </h2>
	<p>Unaggregated series in json: $day.outTemp.series.json():</p>
    <pre>[[1283497200, 1283499000, 46.835437], [1283499000, 1283500800, 44.855907], [1283500800, 1283502600, 43.118351], [1283502600, 1283504400, null], [1283504400, 1283506200, 40.482569], [1283506200, 1283508000, 39.629011], [1283508000, 1283509800, 39.105996], [1283509800, 1283511600, 38.922255], [1283511600, 1283513400, 39.080718], [1283513400, 1283515200, 39.578457], [1283515200, 1283517000, 40.406738], [1283517000, 1283518800, 41.551174], [1283518800, 1283520600, 42.991965], [1283520600, 1283522400, 44.704245], [1283522400, 1283524200, 46.658497], [1283524200, 1283526000, 48.821069], [1283526000, 1283527800, 51.154741], [1283527800, 1283529600, 53.619368], [1283529600, 1283531400, 56.172561], [1283531400, 1283533200, 58.770418], [1283533200, 1283535000, 61.368273], [1283535000, 1283536800, 63.921459]]</pre>

	<p>Unaggregated series, in json with just start times: $day.outTemp.series.json(time_series='start'):</p>
    <pre>[[1283497200, 46.835437], [1283499000, 44.855907], [1283500800, 43.118351], [1283502600, null], [1283504400, 40.482569], [1283506200, 39.629011], [1283508000, 39.105996], [1283509800, 38.922255], [1283511600, 39.080718], [1283513400, 39.578457], [1283515200, 40.406738], [1283517000, 41.551174], [1283518800, 42.991965], [1283520600, 44.704245], [1283522400, 46.658497], [1283524200, 48.821069], [1283526000, 51.154741], [1283527800, 53.619368], [1283529600, 56.172561], [1283531400, 58.770418], [1283533200, 61.368273], [1283535000, 63.921459]]</pre>

	<p>Unaggregated series, in json with start times in milliseconds: $day.outTemp.series.json(time_series='start', time_unit='unix_epoch_ms'):</p>
    <pre>[[1283497200000.0, 46.835437], [1283499000000.0, 44.855907], [1283500800000.0, 43.118351], [1283502600000.0, null], [1283504400000.0, 40.482569], [1283506200000.0, 39.629011], [1283508000000.0, 39.105996], [1283509800000.0, 38.922255], [1283511600000.0, 39.080718], [1283513400000.0, 39.578457], [1283515200000.0, 40.406738], [1283517000000.0, 41.551174], [1283518800000.0, 42.991965], [1283520600000.0, 44.704245], [1283522400000.0, 46.658497], [1283524200000.0, 48.821069], [1283526000000.0, 51.154741], [1283527800000.0, 53.619368], [1283529600000.0, 56.172561], [1283531400000.0, 58.770418], [1283533200000.0, 61.368273], [1283535000000.0, 63.921459]]</pre>

    <p>Unaggregated series, in json, in degrees C: $day.outTemp.series.degree_C.json</p>
    <pre>[[1283497200, 1283499000, 8.241909444444444], [1283499000, 1283500800, 7.142170555555556], [1283500800, 1283502600, 6.176861666666665], [1283502600, 1283504400, null], [1283504400, 1283506200, 4.712538333333332], [1283506200, 1283508000, 4.238339444444444], [1283508000, 1283509800, 3.947775555555554], [1283509800, 1283511600, 3.8456972222222223], [1283511600, 1283513400, 3.9337322222222206], [1283513400, 1283515200, 4.210253888888889], [1283515200, 1283517000, 4.670409999999999], [1283517000, 1283518800, 5.30620777777778], [1283518800, 1283520600, 6.106647222222223], [1283520600, 1283522400, 7.057913888888889], [1283522400, 1283524200, 8.143609444444444], [1283524200, 1283526000, 9.345038333333333], [1283526000, 1283527800, 10.641522777777778], [1283527800, 1283529600, 12.010760000000001], [1283529600, 1283531400, 13.429200555555557], [1283531400, 1283533200, 14.872454444444445], [1283533200, 1283535000, 16.315707222222223], [1283535000, 1283536800, 17.734143888888887]]</pre>

    <p>Unaggregated series, in json, in degrees C, rounded to 2 decimal places: $day.outTemp.series.degree_C.json(ndigits=2)</p>
    <pre>[[1283497200, 1283499000, 8.24], [1283499000, 1283500800, 7.14], [1283500800, 1283502600, 6.18], [1283502600, 1283504400, null], [1283504400, 1283506200, 4.71], [1283506200, 1283508000, 4.24], [1283508000, 1283509800, 3.95], [1283509800, 1283511600, 3.85], [1283511600, 1283513400, 3.93], [1283513400, 1283515200, 4.21], [1283515200, 1283517000, 4.67], [1283517000, 1283518800, 5.31], [1283518800, 1283520600, 6.11], [1283520600, 1283522400, 7.06], [1283522400, 1283524200, 8.14], [1283524200, 1283526000, 9.35], [1283526000, 1283527800, 10.64], [1283527800, 1283529600, 12.01], [1283529600, 1283531400, 13.43], [1283531400, 1283533200, 14.87], [1283533200, 1283535000, 16.32], [1283535000, 1283536800, 17.73]]</pre>

    <p>Unaggregated series, as a formatted string (<em>not JSON</em>) : $day.outTemp.series:</p>
    <pre>00:00, 00:30, 46.8&#176;F
00:30, 01:00, 44.9&#176;F
01:00, 01:30, 43.1&#176;F
01:30, 02:00,    N/A
02:00, 02:30, 40.5&#176;F
02:30, 03:00, 39.6&#176;F
03:00, 03:30, 39.1&#176;F
03:30, 04:00, 38.9&#176;F
04:00, 04:30, 39.1&#176;F
04:30, 05:00, 39.6&#176;F
05:00, 05:30, 40.4&#176;F
05:30, 06:00, 41.6&#176;F
06:00, 06:30, 43.0&#176;F
06:30, 07:00, 44.7&#176;F
07:00, 07:30, 46.7&#176;F
07:30, 08:00, 48.8&#176;F
08:00, 08:30, 51.2&#176;F
08:30, 09:00, 53.6&#176;F
09:00, 09:30, 56.2&#176;F
09:30, 10:00, 58.8&#176;F
10:00, 10:30, 61.4&#176;F
10:30, 11:00, 63.9&#176;F</pre>

    <p>Unaggregated series, by column, as a formatted string (<em>not JSON</em>) $day.outTemp.series.format(order_by='column'):</p>
    <pre>00:00, 00:30, 01:00, 01:30, 02:00, 02:30, 03:00, 03:30, 04:00, 04:30, 05:00, 05:30, 06:00, 06:30, 07:00, 07:30, 08:00, 08:30, 09:00, 09:30, 10:00, 10:30
00:30, 01:00, 01:30, 02:00, 02:30, 03:00, 03:30, 04:00, 04:30, 05:00, 05:30, 06:00, 06:30, 07:00, 07:30, 08:00, 08:30, 09:00, 09:30, 10:00, 10:30, 11:00
46.8&#176;F, 44.9&#176;F, 43.1&#176;F,    N/A, 40.5&#176;F, 39.6&#176;F, 39.1&#176;F, 38.9&#176;F, 39.1&#176;F, 39.6&#176;F, 40.4&#176;F, 41.6&#176;F, 43.0&#176;F, 44.7&#176;F, 46.7&#176;F, 48.8&#176;F, 51.2&#176;F, 53.6&#176;F, 56.2&#176;F, 58.8&#176;F, 61.4&#176;F, 63.9&#176;F</pre>

	<hr/>
	<h2>Aggregated series</h2>
	<p>Aggregated series: $month.outTemp.series(aggregation_type='max', aggregation_interval=86400).json():</p>
    <pre>[[1283324400, 1283410800, 79.825823], [1283410800, 1283497200, 79.224876], [1283497200, 1283583600, 63.921459]]</pre>

	<p>Using shortcut 'day': $month.outTemp.series(aggregation_type='max', aggregation_interval='day').json():</p>
    <pre>[[1283324400, 1283410800, 79.825823], [1283410800, 1283497200, 79.224876], [1283497200, 1283583600, 63.921459]]</pre>

	<p>Order by column: $month.outTemp.series(aggregation_type='max', aggregation_interval=86400).json(order_by="column"):</p>
    <pre>[[1283324400, 1283410800, 1283497200], [1283410800, 1283497200, 1283583600], [79.825823, 79.224876, 63.921459]]</pre>

    <p>Aggregated series, using $jsonize(), and rounding of raw data:
    $jsonize($zip($min.start.unix_epoch_ms.raw, $min.data.degree_C.raw(2), $max.data.degree_C.raw(2)))
    <pre>
    [[1283324400000.0, 4.51, 26.57], [1283410800000.0, 4.18, 26.24], [1283497200000.0, 3.85, 17.73]]
    </pre>


    <hr/>
    <h2>Complex conversions</h2>

    <p>Starting series: $month.windvec.series(aggregation_type='max', aggregation_interval='day').json()</p>
    <pre>
    [[1283324400, 1283410800, [-0.6112556182879079, 9.325958562961368]], [1283410800, 1283497200, [19.935813307366608, 1.3066622382029147]], [1283497200, 1283583600, [20.0, 0.0]], [1283583600, 1283670000, null], [1283670000, 1283756400, null], [1283756400, 1283842800, null], [1283842800, 1283929200, null], [1283929200, 1284015600, null], [1284015600, 1284102000, null], [1284102000, 1284188400, null], [1284188400, 1284274800, null], [1284274800, 1284361200, null], [1284361200, 1284447600, null], [1284447600, 1284534000, null], [1284534000, 1284620400, null], [1284620400, 1284706800, null], [1284706800, 1284793200, null], [1284793200, 1284879600, null], [1284879600, 1284966000, null], [1284966000, 1285052400, null], [1285052400, 1285138800, null], [1285138800, 1285225200, null], [1285225200, 1285311600, null], [1285311600, 1285398000, null], [1285398000, 1285484400, null], [1285484400, 1285570800, null], [1285570800, 1285657200, null], [1285657200, 1285743600, null], [1285743600, 1285830000, null], [1285830000, 1285916400, null]]
    </pre>

    <p> X-component: $month.windvec.series(aggregation_type='max', aggregation_interval='day').x.json()</p>
    <pre>
    [[1283324400, 1283410800, -0.6112556182879079], [1283410800, 1283497200, 19.935813307366608], [1283497200, 1283583600, 20.0], [1283583600, 1283670000, null], [1283670000, 1283756400, null], [1283756400, 1283842800, null], [1283842800, 1283929200, null], [1283929200, 1284015600, null], [1284015600, 1284102000, null], [1284102000, 1284188400, null], [1284188400, 1284274800, null], [1284274800, 1284361200, null], [1284361200, 1284447600, null], [1284447600, 1284534000, null], [1284534000, 1284620400, null], [1284620400, 1284706800, null], [1284706800, 1284793200, null], [1284793200, 1284879600, null], [1284879600, 1284966000, null], [1284966000, 1285052400, null], [1285052400, 1285138800, null], [1285138800, 1285225200, null], [1285225200, 1285311600, null], [1285311600, 1285398000, null], [1285398000, 1285484400, null], [1285484400, 1285570800, null], [1285570800, 1285657200, null], [1285657200, 1285743600, null], [1285743600, 1285830000, null], [1285830000, 1285916400, null]]
    </pre>

    <p> x-component, in knots: $month.windvec.series(aggregation_type='max', aggregation_interval='day').x.knot.json()</p>
    <pre>
    [[1283324400, 1283410800, -0.5311666100812127], [1283410800, 1283497200, 17.323748129049026], [1283497200, 1283583600, 17.379524840000002], [1283583600, 1283670000, null], [1283670000, 1283756400, null], [1283756400, 1283842800, null], [1283842800, 1283929200, null], [1283929200, 1284015600, null], [1284015600, 1284102000, null], [1284102000, 1284188400, null], [1284188400, 1284274800, null], [1284274800, 1284361200, null], [1284361200, 1284447600, null], [1284447600, 1284534000, null], [1284534000, 1284620400, null], [1284620400, 1284706800, null], [1284706800, 1284793200, null], [1284793200, 1284879600, null], [1284879600, 1284966000, null], [1284966000, 1285052400, null], [1285052400, 1285138800, null], [1285138800, 1285225200, null], [1285225200, 1285311600, null], [1285311600, 1285398000, null], [1285398000, 1285484400, null], [1285484400, 1285570800, null], [1285570800, 1285657200, null], [1285657200, 1285743600, null], [1285743600, 1285830000, null], [1285830000, 1285916400, null]]
    </pre>

    <p>Y-component: $month.windvec.series(aggregation_type='max', aggregation_interval='day').y.json()</p>
    <pre>
    [[1283324400, 1283410800, 9.325958562961368], [1283410800, 1283497200, 1.3066622382029147], [1283497200, 1283583600, 0.0], [1283583600, 1283670000, null], [1283670000, 1283756400, null], [1283756400, 1283842800, null], [1283842800, 1283929200, null], [1283929200, 1284015600, null], [1284015600, 1284102000, null], [1284102000, 1284188400, null], [1284188400, 1284274800, null], [1284274800, 1284361200, null], [1284361200, 1284447600, null], [1284447600, 1284534000, null], [1284534000, 1284620400, null], [1284620400, 1284706800, null], [1284706800, 1284793200, null], [1284793200, 1284879600, null], [1284879600, 1284966000, null], [1284966000, 1285052400, null], [1285052400, 1285138800, null], [1285138800, 1285225200, null], [1285225200, 1285311600, null], [1285311600, 1285398000, null], [1285398000, 1285484400, null], [1285484400, 1285570800, null], [1285570800, 1285657200, null], [1285657200, 1285743600, null], [1285743600, 1285830000, null], [1285830000, 1285916400, null]]
    </pre>

    <p>magnitude: $month.windvec.series(aggregation_type='max', aggregation_interval='day').magnitude.json()</p>
    <pre>
    [[1283324400, 1283410800, 9.345969], [1283410800, 1283497200, 19.978589], [1283497200, 1283583600, 20.0], [1283583600, 1283670000, null], [1283670000, 1283756400, null], [1283756400, 1283842800, null], [1283842800, 1283929200, null], [1283929200, 1284015600, null], [1284015600, 1284102000, null], [1284102000, 1284188400, null], [1284188400, 1284274800, null], [1284274800, 1284361200, null], [1284361200, 1284447600, null], [1284447600, 1284534000, null], [1284534000, 1284620400, null], [1284620400, 1284706800, null], [1284706800, 1284793200, null], [1284793200, 1284879600, null], [1284879600, 1284966000, null], [1284966000, 1285052400, null], [1285052400, 1285138800, null], [1285138800, 1285225200, null], [1285225200, 1285311600, null], [1285311600, 1285398000, null], [1285398000, 1285484400, null], [1285484400, 1285570800, null], [1285570800, 1285657200, null], [1285657200, 1285743600, null], [1285743600, 1285830000, null], [1285830000, 1285916400, null]]
    </pre>

    <p>direction: $month.windvec.series(aggregation_type='max', aggregation_interval='day').direction.json()</p>
    <pre>
    [[1283324400, 1283410800, 356.25], [1283410800, 1283497200, 86.25], [1283497200, 1283583600, 90.0], [1283583600, 1283670000, null], [1283670000, 1283756400, null], [1283756400, 1283842800, null], [1283842800, 1283929200, null], [1283929200, 1284015600, null], [1284015600, 1284102000, null], [1284102000, 1284188400, null], [1284188400, 1284274800, null], [1284274800, 1284361200, null], [1284361200, 1284447600, null], [1284447600, 1284534000, null], [1284534000, 1284620400, null], [1284620400, 1284706800, null], [1284706800, 1284793200, null], [1284793200, 1284879600, null], [1284879600, 1284966000, null], [1284966000, 1285052400, null], [1285052400, 1285138800, null], [1285138800, 1285225200, null], [1285225200, 1285311600, null], [1285311600, 1285398000, null], [1285398000, 1285484400, null], [1285484400, 1285570800, null], [1285570800, 1285657200, null], [1285657200, 1285743600, null], [1285743600, 1285830000, null], [1285830000, 1285916400, null]]
    </pre>

    <p>polar: $month.windvec.series(aggregation_type='max', aggregation_interval='day').polar.json()</p>
    <pre>
    [[1283324400, 1283410800, [9.345969, 356.25]], [1283410800, 1283497200, [19.978589, 86.25]], [1283497200, 1283583600, [20.0, 90.0]], [1283583600, 1283670000, null], [1283670000, 1283756400, null], [1283756400, 1283842800, null], [1283842800, 1283929200, null], [1283929200, 1284015600, null], [1284015600, 1284102000, null], [1284102000, 1284188400, null], [1284188400, 1284274800, null], [1284274800, 1284361200, null], [1284361200, 1284447600, null], [1284447600, 1284534000, null], [1284534000, 1284620400, null], [1284620400, 1284706800, null], [1284706800, 1284793200, null], [1284793200, 1284879600, null], [1284879600, 1284966000, null], [1284966000, 1285052400, null], [1285052400, 1285138800, null], [1285138800, 1285225200, null], [1285225200, 1285311600, null], [1285311600, 1285398000, null], [1285398000, 1285484400, null], [1285484400, 1285570800, null], [1285570800, 1285657200, null], [1285657200, 1285743600, null], [1285743600, 1285830000, null], [1285830000, 1285916400, null]]
    </pre>

    <p>polar in knots: $month.windvec.series(aggregation_type='max', aggregation_interval='day').knot.polar.json()</p>
    <pre>
    [[1283324400, 1283410800, [8.121425019468496, 356.25]], [1283410800, 1283497200, [17.360919189682537, 86.25]], [1283497200, 1283583600, [17.379524840000002, 90.0]], [1283583600, 1283670000, null], [1283670000, 1283756400, null], [1283756400, 1283842800, null], [1283842800, 1283929200, null], [1283929200, 1284015600, null], [1284015600, 1284102000, null], [1284102000, 1284188400, null], [1284188400, 1284274800, null], [1284274800, 1284361200, null], [1284361200, 1284447600, null], [1284447600, 1284534000, null], [1284534000, 1284620400, null], [1284620400, 1284706800, null], [1284706800, 1284793200, null], [1284793200, 1284879600, null], [1284879600, 1284966000, null], [1284966000, 1285052400, null], [1285052400, 1285138800, null], [1285138800, 1285225200, null], [1285225200, 1285311600, null], [1285311600, 1285398000, null], [1285398000, 1285484400, null], [1285484400, 1285570800, null], [1285570800, 1285657200, null], [1285657200, 1285743600, null], [1285743600, 1285830000, null], [1285830000, 1285916400, null]]
    </pre>

  <hr/>
  <h2>Iterate over an aggregated series</h2>
  <pre>
    #for ($start, $stop, $data) in $month.outTemp.series(aggregation_type='max', aggregation_interval='day')
      ...
    #end for
  </pre>

    <table>
      <tr>
        <td>Start date</td>
        <td>Stop date</td>
        <td>Max temperature</td>
      </tr>
      <tr>
        <td>2010-09-01</td>
        <td>2010-09-02</td>
        <td>79.83&#176;F</td>
      </tr>
      <tr>
        <td>2010-09-02</td>
        <td>2010-09-03</td>
        <td>79.22&#176;F</td>
      </tr>
      <tr>
        <td>2010-09-03</td>
        <td>2010-09-04</td>
        <td>63.92&#176;F</td>
      </tr>
    </table>

  </body>

</html>
